### PR TITLE
Spinor norms

### PR DESCRIPTION
**Pull request by simonbrandhorst**
More translation of sage code of mine.

### CODE REVIEW

File: src/Groups/matrices/form_group.jl [hunk 1]
```diff
@@ -652,24 +652,15 @@ Return the group of isometries of the lattice `L`.
...
-      # The following can be removed once we have Hecke >= 0.10.26
-      d = degree(L)
-      K = base_field(L)
-      gens = [identity_matrix(K,d)]
-   else
-      gens = Hecke.automorphism_group_generators(L)
+   z = get_attribute(L, :isometry_group)
+   if z !== nothing
+     T = elem_type(base_field(L))
+     return z::MatrixGroup{T, dense_matrix_type(T)}
```
Comment (fingolfin) [inline src/Groups/matrices/form_group.jl]: Maybe use `@attr`? This *might* work, though it is of course not very nice to read:
```
@attr MatrixGroup{elem_type(base_field(L)), dense_matrix_type(elem_type(base_field(L)))}
```

Comment (simonbrandhorst) [reply src/Groups/matrices/form_group.jl]: O.K.

**Comment by simonbrandhorst** [comment]
To pass the tests this needs a recent fix in Hecke.

File: src/Groups/spinor_norms.jl [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+  @assert isone(f)
+  return spinor_norm[1,1]
+end
+
+function valuation(A::fmpq_mat, p)
+  return minimum([a==0 ? PosInf() : valuation(a, p) for a in A])
+end
+
+"""
+  det_spin(G::fmpq_mat, T::fmpq_mat, p, nu) -> Tuple{fmpq, fmpq}
```
Comment (thofma) [inline src/Groups/spinor_norms.jl]:
```suggestion
    det_spin(G::fmpq_mat, T::fmpq_mat, p, nu) -> Tuple{fmpq, fmpq}
```

File: src/Groups/spinor_norms.jl [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+      end
+    end
+  end
+  Agap, i1,_ = _isomorphic_gap_group(D)
+  return hom(Oq,Agap,gens(Oq),[i1(proj(result[f])) for f in gens(Oq)],check=false)
+end
+
+
+"""
+image_in_Oq(L::ZLat) -> AutomorphismGroup{Hecke.TorQuadMod}, GAPGroupHomomorphism
```
Comment (thofma) [inline src/Groups/spinor_norms.jl]:
```suggestion
    image_in_Oq(L::ZLat) -> AutomorphismGroup{Hecke.TorQuadMod}, GAPGroupHomomorphism
```

File: docs/oscar_references.bib [hunk 4]
```diff
@@ -754,6 +754,20 @@ @Book{PZ97
   pages         = {xiv+499}
 }
 
+@article{[Shim2018],
```
Comment (thofma) [inline docs/oscar_references.bib]:
```suggestion
@article{Shim2018,
```
?

**Pull request merged**